packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -581,8 +581,9 @@ class DevFS {
     }
     assert((file is Link) || (file is File));
     if (ignoreDotFiles && fs.path.basename(file.path).startsWith('.')) {
-      // Skip dot files.
-      return true;
+      // Skip dot files, but not the '.packages' file (even though in dart1
+      // mode devfs['.packages'] will be overwritten with synthesized string content).
+      return fs.path.basename(file.path) != '.packages';
```

Comment:
fs.path.basename(file.path) is called twice, does it make sense to hoist that into a local variable?

Reply:
Done